mimic/test/test_loadbalancer.py

```diff
@@ -733,3 +851,49 @@ def test_delete_load_balancer_and_pending_delete_state(self):
             self, self.root, "GET", self.uri + '/loadbalancers/' + str(lb["id"])
             + '/nodes')
         self.assertEqual(self.successResultOf(list_nodes).code, 404)
+
+    def test_bulk_delete_empty_list_takes_precedence_over_immutable(self):
+        """
+        When bulk deleting no nodes, the error indicating nodes must be specified
+        is returned even when the LB is not ACTIVE.
+        """
+        metadata = [{"key": "lb_pending_update", "value": 30}]
+        lb = self._create_loadbalancer(metadata)
+
+        # Add a node, which should put it into PENDING-UPDATE
+        create_node_response = self._add_node_to_lb(lb["id"])
+        self.assertEqual(create_node_response.code, 202)
+        updated_lb = self._get_loadbalancer(lb["id"])
+        self.assertEqual(updated_lb["loadBalancer"]["status"], "PENDING-UPDATE")
+
+        # Now, trying to bulk-delete an empty list of nodes will still return
+        # the empty-nodes error.
+        response, body = _bulk_delete(self, self.root, self.uri, lb['id'], [])
+        self.assertEqual(response.code, 400)
+        self.assertEqual(
+            body,
+            {'code': 400,
+             'message': "Must supply one or more id's to process this request."})
+
+    def test_bulk_delete_not_active(self):
+        """
+        When bulk deleting nodes while the LB is not ACTIVE, a special error is
+        returned, even when some of the nodes are invalid.
+        """
+        metadata = [{"key": "lb_pending_update", "value": 30}]
+        lb = self._create_loadbalancer(metadata)
+
+        # Add a node, which should put it into PENDING-UPDATE
+        create_node_response = self._add_node_to_lb(lb["id"])
+        self.assertEqual(create_node_response.code, 202)
+        updated_lb = self._get_loadbalancer(lb["id"])
+        self.assertEqual(updated_lb["loadBalancer"]["status"], "PENDING-UPDATE")
+
+        # Now, trying to bulk-delete nodes (including invalid ones) will cause
+        # it to return the special error
+        response, body = _bulk_delete(
+            self, self.root, self.uri, lb['id'], [100, 200])
+        self.assertEqual(response.code, 422)
+        self.assertEqual(
+            body,
+            {u'message': u'LoadBalancer is not ACTIVE', u'code': 422})
```

Comment:
This case doesn't even pass a valid node ID at all to the deletion request, so I don't think it's useful. Also there's already a test case that ensures that passing some valid and some invalid IDs to a bulk deletion doesn't delete the valid ones.